src/main/java/cms/gov/madie/measure/services/MeasureService.java

```diff
@@ -392,4 +392,26 @@ protected void updateReferenceId(MeasureMetaData metaData) {
       metaData.setReferences(references);
     }
   }
+
+  public List<Measure> findAllByMeasureSetId(String measureSetId) {
+    List<Measure> measures = measureRepository.findAllByMeasureSetId(measureSetId);
+    return measures;
+  }
+
+  public void deleteVersionedMeasures(List<Measure> measures) {
+    List<Measure> versionedMeasures =
+        measures.stream()
+            .filter(
+                measure ->
+                    measure.getMeasureMetaData() != null && !measure.getMeasureMetaData().isDraft())
+            .collect(Collectors.toList());
+    if (!CollectionUtils.isEmpty(versionedMeasures)) {
+      String deletedMeasureIds =
+          versionedMeasures.stream()
+              .map(measure -> measure.getId())
+              .collect(Collectors.joining(","));
+      measureRepository.deleteAll(versionedMeasures);
```

Comment:
I'm impressed this works with a comma delimited String. The interface says it accepts an Iterable, and I did not consider a String with commas an Iterable 🤣 .

Reply:
@jkotanchik-SB The comma delimited string (deletedMeasureIds) is only for the logging below at line 414. For delete on line 413 it's a list of measures (versionedMeasures), not the ids.